dblookupext/esdtSupply/corrections.go

```diff
@@ -0,0 +1,136 @@
+package esdtSupply
+
+import (
+	"bytes"
+	"errors"
+	"math/big"
+
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/storage"
+)
+
+const (
+	addOperation      = "add"
+	subtractOperation = "subtract"
+
+	supplyCorrectionPrefix = "supplyCorrection"
+)
+
+type supplyCorrectionProcessor struct {
+	shardID  uint32
+	logsProc *logsProcessor
+}
+
+func newSupplyCorrectionProcessor(shardID uint32, logsProc *logsProcessor) *supplyCorrectionProcessor {
+	return &supplyCorrectionProcessor{
+		shardID:  shardID,
+		logsProc: logsProc,
+	}
+}
+
+func (scp *supplyCorrectionProcessor) applySupplyCorrections(supplyCorrections []config.SupplyCorrection) error {
+	supplies := make(map[string]*SupplyESDT)
+	for _, supplyCorrection := range supplyCorrections {
+		shouldApply, err := scp.shouldApplyCorrection(supplyCorrection)
+		if err != nil {
+			return err
+		}
+		if !shouldApply {
+			continue
+		}
+
+		tokenSupply, err := scp.logsProc.getESDTSupply([]byte(supplyCorrection.Token))
+		if err != nil {
+			return err
+		}
+
+		// if token supply was recomputed should not apply supply correction
+		if tokenSupply.RecomputedSupply {
+			return nil
+		}
+
+		correctionValue, ok := big.NewInt(0).SetString(supplyCorrection.Value, 10)
+		if !ok {
+			return errors.New("failed to parse supplyCorrection value")
+		}
+
+		switch supplyCorrection.Operation {
+		case addOperation:
+			tokenSupply.Supply.Add(tokenSupply.Supply, correctionValue)
```

Comment:
seems like no need for the type of operation add, or subtract, if we allow both negative values and positive values.
then depending on the sign of the value we need to apply either mint or burn

Also we could probably use a check not to be left with negative supply (on burn) and log a warning if so.

Reply:
yes, cleaner without an operation... we cand have only add or subtract, right? that means positive or negative value in the correction

Reply:
removed the `operation` field. 
and also these settings from `config.toml`